scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/AzureBlobDownloader.java

```diff
@@ -106,19 +115,19 @@ public Serializable execute(TaskResult... results) throws IOException {
         //download a single blob
         if (optionalBlobName.isPresent()) {
             //check weather or not the outputPath is a folder path or a file path
-            if (outputPath.lastIndexOf('/') == outputPath.length() - 1) {
-                createDirIfNotExists(file);
-                String azureBlobLocalRelativePath = Paths.get(outputPath,
-                                                              Paths.get(optionalBlobName.get())
-                                                                   .getFileName()
-                                                                   .toString())
-                                                         .toString();
+            String azureBlobLocalRelativePath = Paths.get(outputPath,
+                                                          Paths.get(optionalBlobName.get()).getFileName().toString())
+                                                     .toString();
+            if (outputPath.endsWith(File.separator)) {
+                SchedulerExamplesUtils.createDirIfNotExists(file);
+                downloadBlob(new File(azureBlobLocalRelativePath), optionalBlobName.get());
+            } else if (outputPath.equals(getLocalSpace())) {
                 downloadBlob(new File(azureBlobLocalRelativePath), optionalBlobName.get());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "optionalBlobName.isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)